pandas/core/series.py

```diff
@@ -2324,9 +2324,11 @@ def combine(self, other, func, fill_value=None):
         elif is_extension_array_dtype(self.values):
             # The function can return something of any type, so check
             # if the type is compatible with the calling EA
+            # ExtensionArray._from_sequence can raise anything, so we
+            # have to catch everything.
             try:
                 new_values = self._values._from_sequence(new_values)
-            except TypeError:
+            except Exception:
```

Comment:
Can you add a comment here linking to this PR.  Inevitably I'm going to forget why this isn't catching something more specific.